app/knight_stats.py

```diff
@@ -0,0 +1,33 @@
+class Knight:
+    def __init__(self, equipment: dict) -> None:
+        self.name = equipment.get("name")
+        self.hp = equipment.get("hp")
+        self.power = equipment.get("power")
+        self.protection = 0
+        self.armour = equipment.get("armour")
+        self.weapon = equipment.get("weapon")
+        self.potion = equipment.get("potion")
+
+    def apply_potion(self) -> None:
+        for stat, value in self.potion.get("effect").items():
```

Comment:
Let's imagine that potion doesn't have an "effect" key. Then `self.potion.get("effect")` will give you None. And it will lead to `None.items()` --> `AttributeError: 'NoneType' object has no attribute 'items'`
So It will be much more readable and easier to debug if you will access by key, without using `get`

Reply:
Fix everywhere